core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2730,10 +2730,30 @@ object ZIO {
    */
   def reduceAllPar[R, R1 <: R, E, A](a: ZIO[R, E, A], as: Iterable[ZIO[R1, E, A]])(
     f: (A, A) => A
-  ): ZIO[R1, E, A] =
-    as.foldLeft[ZIO[R1, E, A]](a) { (l, r) =>
-      l.zipPar(r).map(f.tupled)
-    }
+  ): ZIO[R1, E, A] = {
+    def prepend(a: ZIO[R, E, A], as: Iterable[ZIO[R1, E, A]]): Iterable[ZIO[R1, E, A]] =
+      new Iterable[ZIO[R1, E, A]] {
+        override def iterator: Iterator[ZIO[R1, E, A]] = new Iterator[ZIO[R1, E, A]] {
```

Comment:
Not sure, if we should locate this method here or somewhere else

Reply:
Or maybe, it's OK to just prepend using `Iterable(a) ++ as`

Reply:
This seems unnecessary. You could do `Iterator(a) ++ as.iterator` to be lazy about this.